### PR TITLE
Remove `index.mapping.single_type=false` from core/tests

### PR DESCRIPTION
**Pull request by s1monw**
This change cleans up core tests to not use `index.mapping.single_type=false`
but instead where applicable use a single type or markt the index as created
with a pre 6.x version.

Relates to #24961

### CODE REVIEW

**Review by jpountz — APPROVED** [review]
I think it's a fair way to fix those tests. LGTM.

**Comment by nik9000** [comment]
I dunno. I expected this process to consist more of removing multiple types in the tests and less of declaring indices to be old. The test names don't make it look like they are testing old index behavior. They mostly look like they intend to test "normal" index behavior.

I think this kicks the can down to 7.0 when someone will have to look at these tests and decide weather to rewrite them with one type or just remove them.

**Comment by jpountz** [comment]
> The test names don't make it look like they are testing old index behavior.

They do since they are using parent/child?

**Comment by s1monw** [comment]
> I dunno. I expected this process to consist more of removing multiple types in the tests and less of declaring indices to be old. The test names don't make it look like they are testing old index behavior. They mostly look like they intend to test "normal" index behavior.

I did convert quite a few of them, actually all of the ones that are possible and I invested a lot of time. I think you should look closer and tell me the ones that I missed. We are in-fact supporting multiple types for 5.x indices so we should also test the APIs against it. This is why I kept or duplicated a lot of them.

File: core/src/test/java/org/elasticsearch/indices/stats/IndexStatsIT.java [hunk 1]
```diff
@@ -570,7 +581,7 @@ public void testSegmentsStats() {
 
     public void testAllFlags() throws Exception {
         // rely on 1 replica for this tests
-        assertAcked(prepareCreate("test1").setSettings("index.mapping.single_type", false));
+        assertAcked(prepareCreate("test1").setSettings("index.version.created", Version.V_5_6_0.id));
```
Comment (nik9000) [inline core/src/test/java/org/elasticsearch/indices/stats/IndexStatsIT.java]: Can you add a comment about why it is important that we have indices with multiple types?

File: core/src/test/java/org/elasticsearch/indices/stats/IndexStatsIT.java [hunk 2]
```diff
@@ -692,7 +703,7 @@ public void testFlagOrdinalOrder() {
     }
 
     public void testMultiIndex() throws Exception {
-        assertAcked(prepareCreate("test1").setSettings("index.mapping.single_type", false));
+        assertAcked(prepareCreate("test1").setSettings("index.version.created", Version.V_5_6_0.id));
```
Comment (nik9000) [inline core/src/test/java/org/elasticsearch/indices/stats/IndexStatsIT.java]: Can you add a comment about why it is important that we have indices with multiple types?

File: core/src/test/java/org/elasticsearch/indices/stats/IndexStatsIT.java [hunk 3]
```diff
@@ -732,7 +743,7 @@ public void testMultiIndex() throws Exception {
 
     public void testFieldDataFieldsParam() throws Exception {
         assertAcked(client().admin().indices().prepareCreate("test1")
-                .setSettings("index.mapping.single_type", false)
```
Comment (nik9000) [inline core/src/test/java/org/elasticsearch/indices/stats/IndexStatsIT.java]: Is it worth adding an index with a single type to this?

File: core/src/test/java/org/elasticsearch/indices/stats/IndexStatsIT.java [hunk 4]
```diff
@@ -780,7 +791,7 @@ public void testFieldDataFieldsParam() throws Exception {
 
     public void testCompletionFieldsParam() throws Exception {
         assertAcked(prepareCreate("test1")
-                .setSettings("index.mapping.single_type", false)
+                .setSettings("index.version.created", Version.V_5_6_0.id) // allows for multiple types
```
Comment (nik9000) [inline core/src/test/java/org/elasticsearch/indices/stats/IndexStatsIT.java]: Same.

File: core/src/test/java/org/elasticsearch/indices/template/SimpleIndexTemplateIT.java [hunk 5]
```diff
@@ -383,7 +392,7 @@ public void testIndexTemplateWithAliases() throws Exception {
                 .get();
 
         assertAcked(prepareCreate("test_index")
-                .setSettings("index.mapping.single_type", false)
+                .setSettings("index.version.created", Version.V_5_6_0.id) // allow for multiple version
```
Comment (nik9000) [inline core/src/test/java/org/elasticsearch/indices/template/SimpleIndexTemplateIT.java]: Is it worth adding a single type variant of this?

File: core/src/test/java/org/elasticsearch/search/fetch/subphase/highlight/HighlighterSearchIT.java [hunk 6]
```diff
@@ -1358,14 +1359,13 @@ public void testCommonTermsTermVector() throws IOException {
...
                 .put(indexSettings())
-                .put("index.mapping.single_type", false)
                 .put("index.analysis.analyzer.synonym.tokenizer", "whitespace")
                 .putArray("index.analysis.analyzer.synonym.filter", "synonym", "lowercase")
                 .put("index.analysis.filter.synonym.type", "synonym")
                 .putArray("index.analysis.filter.synonym.synonyms", "quick => fast");
 
-        assertAcked(prepareCreate("test").setSettings(builder.build()).addMapping("type1", type1TermVectorMapping())
-                .addMapping("type2",
+        assertAcked(prepareCreate("test").setSettings(builder.build()).addMapping("type1", type1TermVectorMapping()));
```
Comment (nik9000) [inline core/src/test/java/org/elasticsearch/search/fetch/subphase/highlight/HighlighterSearchIT.java]: Can you rename these indices? I know I'll get `test` and `test1` mixed up. Even `test1` and `test2` would be better.

File: core/src/test/java/org/elasticsearch/search/fields/SearchFieldsIT.java [hunk 7]
```diff
@@ -872,7 +866,7 @@ public void testScriptFields() throws Exception {
 
     public void testLoadMetadata() throws Exception {
         assertAcked(prepareCreate("test")
-                .setSettings("index.mapping.single_type", false)
+                .setSettings("index.version.created", Version.V_5_6_0.id)
```
Comment (nik9000) [inline core/src/test/java/org/elasticsearch/search/fields/SearchFieldsIT.java]: If we have another test for routing I'd rename this one to `testLoadParent` or something. Then we can nuke it in 7.0 easilly.

File: core/src/test/java/org/elasticsearch/search/query/SearchQueryIT.java [hunk 8]
```diff
@@ -1181,7 +1191,36 @@ public void testTermsLookupFilter() throws Exception {
...
+        searchResponse = client().prepareSearch().setQuery(idsQuery(Strings.EMPTY_ARRAY).addIds("1")).get();
+        assertHitCount(searchResponse, 1L);
+        assertThat(searchResponse.getHits().getHits().length, equalTo(1));
+
+        searchResponse = client().prepareSearch().setQuery(idsQuery("type1", "type2", "doc").addIds("1", "2", "3", "4")).get();
+        assertHitCount(searchResponse, 3L);
+        assertThat(searchResponse.getHits().getHits().length, equalTo(3));
+    }
+
+    public void testBasicQueryByIdMultiType() throws Exception {
```
Comment (nik9000) [inline core/src/test/java/org/elasticsearch/search/query/SearchQueryIT.java]: 👍

**Review by martijnvg — APPROVED** [review]
LGTM

File: core/src/test/java/org/elasticsearch/update/UpdateIT.java [hunk 9]
```diff
@@ -461,7 +462,7 @@ public void testUpdateRequestWithScriptAndShouldUpsertDoc() throws Exception {
 
     public void testContextVariables() throws Exception {
         assertAcked(prepareCreate("test")
-                        .setSettings("index.mapping.single_type", false)
+                        .setSettings("index.version.created", Version.V_5_6_0.id)
```
Comment (nik9000) [inline core/src/test/java/org/elasticsearch/update/UpdateIT.java]: Maybe worth leaving a comment that we can remove this setting, the parent, and the second document once we no longer support types.

**Review by nik9000 — COMMENTED** [review]
I left a bunch of comments, mostly asking for comments explaining why we need multiple types and/or asking whether we should have a "single type" variant of tests.

File: core/src/test/java/org/elasticsearch/indices/mapping/UpdateMappingIntegrationIT.java [hunk 10]
```diff
@@ -62,14 +67,19 @@
...
+
     public void testDynamicUpdates() throws Exception {
         client().admin().indices().prepareCreate("test")
                 .setSettings(
                         Settings.builder()
                                 .put("index.number_of_shards", 1)
                                 .put("index.number_of_replicas", 0)
                                 .put(MapperService.INDEX_MAPPING_TOTAL_FIELDS_LIMIT_SETTING.getKey(), Long.MAX_VALUE)
-                                .put("index.mapping.single_type", false)
+                                .put("index.version.created", Version.V_5_6_0) // for multiple types
```
Comment (s1monw) [inline core/src/test/java/org/elasticsearch/indices/mapping/UpdateMappingIntegrationIT.java]: no I don't think so

File: core/src/test/java/org/elasticsearch/indices/stats/IndexStatsIT.java [hunk 11]
```diff
@@ -378,7 +389,7 @@ public void testThrottleStats() throws Exception {
     }
 
     public void testSimpleStats() throws Exception {
-        assertAcked(prepareCreate("test1").setSettings("index.mapping.single_type", false));
+        assertAcked(prepareCreate("test1").setSettings("index.version.created", Version.V_5_6_0.id)); // allows for multiple types
```
Comment (s1monw) [inline core/src/test/java/org/elasticsearch/indices/stats/IndexStatsIT.java]: man it's like deleting 5 lines, very simple to `untangle` all the stuff I am doing above is so much duplication for not much win if at all. I thin this one I will just keep

File: core/src/test/java/org/elasticsearch/update/UpdateIT.java (hunk 9 above)
Comment (s1monw) [inline core/src/test/java/org/elasticsearch/update/UpdateIT.java]: wont' the test just fail in that case?

**Comment by s1monw** [comment]
@nik9000 I pushed changes

**Review by nik9000 — APPROVED** [review]

**Pull request merged**